### ISSUE TITLE
Adopted Hyperparameter Tuning Process for WikiQA

### ISSUE THREAD

**Issue by avoidthevoid** [NTMC-Community/MatchZoo#424]
Hi there,

The default configurations in `examples/WikiQA/config` have been optimized for the WikiQA dataset.

Now, I am however wondering which hyperparameters where included in the hyperparameter tuning so that eventually these parameter configurations were selected? 

1. I am especially wondering why **Duet** uses the `embed_glove_d50_norm` embedding and why e.g. **ANMM** uses the `embed.idf` embedding and **MatchPyramid** uses the `embed_glove_d300_norm` embedding? Did you test every embedding for each model and selected the best-performing config?
2. Apart from that, I would also be curious to hear about the **learning rate**, **optimizer** and **dropout rates** that were considered in the hyperparameter tuning process?
3. Finally, there seem to be some differences between the **architecture** of Duet as specified in [the original paper ](https://dl.acm.org/citation.cfm?id=3052579)compared to your implementation (e.g. your implementation does not have a **maxpool** for the query, employs **single instead of double hidden layers** and seems to have **smaller hidden layers** (sized around 30 instead of 300)). Were these changes also part of the tuning process?

Looking forward to your reply, thanks in advance!

**Comment by bwanglzu** [comment]
@avoidthevoid I was not able to answer questions related to the current version of Matchzoo, but I'd suggest you to take a look at [matchzoo 2.0](https://github.com/NTMC-Community/MatchZoo/tree/2.0), it's under development, will be released in about 2 months.

Use matchzoo 2.0 api it's much easier for you to fine-tune the hyper-parameters, an example tutorial can be see at [use matchzoo 2.0 to train a Deep Semantic Structured Model](https://nbviewer.jupyter.org/github/NTMC-Community/MatchZoo/blob/2.0/notebooks/wikiqa/dssm.ipynb).

**Comment by avoidthevoid** [comment]
Hi @bwanglzu,

Thank you for your reply and great job on the work on MatchZoo 2.0! Due to time constraints, I will unfortunately have to stick to the 1.0 version for now. If any of your colleaugues would have the answer to the 3 questions, that would really be wonderful.

Anyhow, thanks for your time!

**Comment by yangliuy** [comment]
@avoidthevoid Thank you for your questions!

For Q1, it is not feasible to test the performance of every different embedding inputs. As you know, we have many different choices on the training algorithms (Skip-gram, CBOW or Glove), different embedding dimensions (100,200,300,...).  Thus it is not possible to try all combinations for this. But there are some common settings adopted by many previous IR/NLP papers, which also prove the effectiveness of these settings. For example, many papers use 300 as the embedding dimension and Glove_840_300d as the input pre-trained word embedding. You can refer to these settings when you run you own experiments to save some time/GPU costs. 

For Q2, similar answers are also applicable for them. We tried to tune as many settings as possible when  we release the benchmark results on WikiQA. But there may still exist better hyper-parameters which are not covered by our experiments.  Feel free to explore more settings if you'd like and you have enough GPU resources/time. I suggest you to refer to some reported hyper-parameters in previous NLP/IR papers to save some time.

For Q3, it is possible. Maybe @faneshion @pl8787 can give you the answer for this question.

Thanks!

**Comment by avoidthevoid** [comment]
Hi @yangliuy,

Thank you for the elaborate answers, they are very helpful. 
I will be on the outlook for the answer to Q3.

Thank you for your efforts!

**Comment by uduse** [comment]
Closed due to inactivity.
